VE-1975: Modified NodeImage to return array of images instead of one image

inez opened the pull request:
Added support for <tabber>
Hero image still works correct

Commit:
VE-1975: Modified NodeImage to return array of images instead of one image

Added support for <tabber>
Hero image still works correct

pauloslund commented:
Seems you were in the middle of adding the description, but other then that the code looks good to me. One thing of note though, it seems on mobile that `$infoboxdata` (https://github.com/Wikia/app/blob/VE-1975/extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php#L46) doesn't actually contain the tabber information.

---

Viewing this page: http://visualeditor.paulo.wikia-dev.com/wiki/1383649460634

## Desktop:

``` php
Array
(
    [0] => Array
        (
            [type] => title
            [data] => Array
                (
                    [value] => Jono
                )

        )

    [1] => Array
        (
            [type] => image
            [data] => Array
                (
                    [0] => Array
                        (
                            [url] => http://vignette.wikia-dev.com/visualeditor/images/2/2a/InezsTest2.jpg/revision/latest?cb=20140922231339
                            [name] => InezsTest2.jpg
                            [key] => InezsTest2.jpg
                            [alt] => Tab One
                            [caption] => Tab One
                        )

                    [1] => Array
                        (
                            [url] => http://vignette.wikia-dev.com/visualeditor/images/b/b8/Challenger.jpg/revision/latest?cb=20140626002212
                            [name] => Challenger.jpg
                            [key] => Challenger.jpg
                            [alt] => Challenger
                            [caption] => Challenger
                        )

                )

        )

)
```

## Mobile:

``` php
Array
(
    [0] => Array
        (
            [type] => title
            [data] => Array
                (
                    [value] => Jono
                )

        )

)
```

Commit:
VE-1975: Added support for mobile (ArticleAsJson) and usage of XML parsing

Special handling for mobile is needed because article for mobile is delivered within ArticleAsJson which makes a lot of transformation during the parsing process (instead of for instance postprocessing).
Parsing HTML with XML parser seems to be giving better results than using just regexes and also it gives as ability to handle nested tabbers in a future in a better way.
